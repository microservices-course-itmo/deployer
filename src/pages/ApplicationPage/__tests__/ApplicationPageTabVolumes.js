import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ApplicationPageTabVolumes } from '../tabs/ApplicationPageTabVolumes/ApplicationPageTabVolumes'
import { List, ListItem, ListItemSecondaryAction, ListItemText, Checkbox } from '@material-ui/core'


Enzyme.configure({ adapter: new Adapter() });


describe('ApplicationPageTabVolumes', ()=>{
    let volumes = ['chiko', 'piko', 'koko']
    beforeEach(()=> {
        volumes = ['chiko', 'piko', 'koko']
    })
    it('renders correctly', () => {
        const CheckVol = renderer
          .create(<ApplicationPageTabVolumes volumes={volumes}/>)
          .toJSON();
        expect(CheckVol).toMatchSnapshot();
      });

      it('Should contain 3 checkbox', () => {
        const wrapper = shallow(<ApplicationPageTabVolumes volumes={volumes}/>);
        expect(wrapper.find(ListItem)).toHaveLength(3);
      });

      it('Should contain right hader', () => {
        const wrapper = shallow(<ApplicationPageTabVolumes volumes={volumes}/>);
        expect(wrapper.find('h3').text()).toEqual('ApplicationPageTabVolumes');
      });
})