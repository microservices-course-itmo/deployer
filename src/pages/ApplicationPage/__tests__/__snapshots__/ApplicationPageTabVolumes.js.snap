// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationPageTabVolumes renders correctly 1`] = `
<div>
  <h3>
    ApplicationPageTabVolumes
  </h3>
  <ul
    className="MuiList-root makeStyles-root-1 MuiList-dense MuiList-padding"
  >
    <li
      className="MuiListItem-container"
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemText-root MuiListItemText-dense"
          id="checkbox-list-secondary-label-chiko"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
          >
            Volume: chiko
          </span>
        </div>
      </div>
      <div
        className="MuiListItemSecondaryAction-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-3 Mui-checked MuiIconButton-colorSecondary MuiIconButton-edgeEnd"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              aria-labelledby="checkbox-list-secondary-label-chiko"
              checked={true}
              className="PrivateSwitchBase-input-5"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-container"
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemText-root MuiListItemText-dense"
          id="checkbox-list-secondary-label-piko"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
          >
            Volume: piko
          </span>
        </div>
      </div>
      <div
        className="MuiListItemSecondaryAction-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-3 Mui-checked MuiIconButton-colorSecondary MuiIconButton-edgeEnd"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              aria-labelledby="checkbox-list-secondary-label-piko"
              checked={true}
              className="PrivateSwitchBase-input-5"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-container"
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemText-root MuiListItemText-dense"
          id="checkbox-list-secondary-label-koko"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-displayBlock"
          >
            Volume: koko
          </span>
        </div>
      </div>
      <div
        className="MuiListItemSecondaryAction-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-checked-3 Mui-checked MuiIconButton-colorSecondary MuiIconButton-edgeEnd"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              aria-labelledby="checkbox-list-secondary-label-koko"
              checked={true}
              className="PrivateSwitchBase-input-5"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
        </span>
      </div>
    </li>
  </ul>
</div>
`;
